test/Table_Tests/src/Database/Upload_Spec.enso

```diff
@@ -321,16 +352,19 @@ spec make_new_connection prefix persistent_connector=True =
 
         Test.specify "should fail if the target table already exists" <|
             name = Name_Generator.random_name "table-already-exists"
-            db_table = connection.create_table name [Column_Description.Value "X" Value_Type.Integer] temporary=True
+            preexisting = connection.create_table name [Column_Description.Value "X" Value_Type.Integer] temporary=True
+            preexisting.update_rows (Table.new [["X", [42]]]) . should_succeed
             t = Table.new [["Y", ['a', 'b']]]
             run_with_and_without_output <|
                 r1 = t.select_into_database_table connection name temporary=True
                 r1.should_fail_with Table_Already_Exists
                 r2 = t.select_into_database_table connection name temporary=False
                 r2.should_fail_with Table_Already_Exists
 
-            db_table.column_names . should_equal ["X"]
-            db_table.at "X" . to_vector . should_equal []
+            # Ensure that the table was not modified.
+            preexisting2 = connection.query name
+            preexisting2.column_names . should_equal ["X"]
+            preexisting2.at "X" . to_vector . should_equal [42]
```

Comment:
Added a few tests to ensure that preexisting data is never affected.

Although they were already passing without the bugfix. But good to be sure we never regress on this.